### ISSUE TITLE
macOS 10.12 Dock icon becomes Question mark after restart

### ISSUE THREAD

**Issue by denis1994** [atom/atom#13695]
### Description

After every restart of macOS 10.12, the Atom dock symbol becomes a question mark and Atom can't be opened.

### Steps to Reproduce

1. Open Atom
2. Select "Keep in dock"
3. Restart

**Expected behavior:** Atom symbol stay in dock normal

**Actual behavior:** Atom symbol becomes question mark and can't be openend

**Reproduces how often:** After every restart

### Versions

Atom 1.13.0

### Additional Information

macOS Sierry 10.12.3 on the new MacBook Pro 15" with TouchBar

**Comment by lee-dohm** [comment]
This has been brought up on [Discuss, the official Atom message board,](https://discuss.atom.io/t/atom-icon-disappearing-from-macos-dock/37622) but we haven't been able to figure out what is specific to your configurations that is causing this problem.  The things I would ask are:

1. How did you install Atom?
1. In what directory did you install Atom?

**Comment by denis1994** [comment]
An Update about it: 

I reinstalled Atom 1.13.1 and now it stays in the dock...

**Comment by lee-dohm** [comment]
It sounds like it is fixed then? If the problem reoccurs, let us know and I'll be happy to reopen the issue so that we can attempt to investigate again.

Thanks for following up!

**Comment by MetaMemoryT** [comment]
1.14.3 still having this problem

👍

**Comment by javed24** [comment]
I installed 1.14.4 on MacOS 10.12.3 and having this issue every time I restart. Tried reinstalling, didn't work. Is there any solution to this issue?

**Comment by Marahin** [comment]
This issue is still happening on my end. Ping @lee-dohm as the issue *should* be reopened.

**Comment by lee-dohm** [comment]
If this is still happening, please _at a bare minimum_ answer the questions I included above otherwise there's really no point to reopening this because we still can't reproduce the problem.

**Comment by Marahin** [comment]
@lee-dohm pardon me, I must have scrolled through without noticing them. 


>How did you install Atom?

Downloaded the .dmg from atom.io if I remember correctly. 

>In what directory did you install Atom?

I'm running macOS so Applications. 

I also noticed that sometimes (when the question mark appears instead of Atom logo) the commandline also does not know what 'atom' is. I can't reproduce right now (as everything works today).

**Comment by lee-dohm** [comment]
> I'm running macOS so Applications.

Is that `/Applications`? Or `~/Applications`?

> I also noticed that sometimes (when the question mark appears instead of Atom logo) the commandline also does not know what 'atom' is.

What is unique about the way that your user account is configured? Are you running as a Guest? Are you running Atom from different user accounts? Did you install Atom as `root`?

**Comment by Marahin** [comment]
I just checked. I remember perfectly that I dragged Atom to Applications in finder window - which one was it or is it I don't know, I am not a long-term macOS user. 
However now when I `cd` to /Applications or ~/Applications I can see no Atom. 

`which atom` returns:
```
➜  / which atom
/usr/local/bin/atom
```

>What is unique about the way that your user account is configured? 
Unsure. My Mac is used with a single-account setup, with zsh and some basic webdev stuff. Nothing extraordinary, and definitely I have not modified the macOS application workflow anyhow (atom is the only application that acts weird).

> Are you running as a Guest?
No.

> Are you running Atom from different user accounts?
No, I'm running it from the same account that I installed it with, and this is the only user account present on the machine.

> Did you install Atom as `root`?

I installed Atom by dragging the icon to Applications in finder. What was going under the hood I don't know, but at the time I was logged in on the only account present on my machine - so my user account. I have however sudo privilege and access that I use from time to time.

**Comment by lee-dohm** [comment]
> However now when I cd to /Applications or ~/Applications I can see no Atom.

This would be the cause of the problem. If I delete the `Atom.app` from `/Applications` (where I had it installed), then I get a question mark on the Dock icon:

<img width="389" alt="screen shot 2017-03-06 at 2 08 12 pm" src="https://cloud.githubusercontent.com/assets/1038121/23632346/5c413206-0276-11e7-9ce0-6f319e8e5660.png">

So I would recommend installing it again.

**Comment by Marahin** [comment]
@lee-dohm but I did not remove it; and, how come it's still working? :) 
I will apply your suggestion and I will provide feedback if something happens again (hopefully: not). 

Thank you.

**Comment by lee-dohm** [comment]
> but I did not remove it; and, how come it's still working? :)

No idea ¯\\\_(ツ)_/¯

**Comment by sstubblefield** [comment]
I'm having this issue too.

I installed it the same way. Steps I took:
- Visit http://atom.io
- Click the big "Download for Mac" button in the Hero area.
- Extract atom-mac.zip into the Downloads folder.
- Open a new Finder window, pointed at `/Applications`.
- Drag Atom.app from Downloads into `/Applications`.
- Drag Atom.app from `/Applications` to Dock.
- Reboot computer.
- Big old question mark where Atom used to be.

**Secondary issues:**  
Before I rebooted the computer, I opened Atom. A second icon appeared at the end of the dock, appearing to be the icon that Atom now runs from. I have two Atom icons in the dock now.

Fun fact, this exact issue happens with GitKraken - another Electron app. I have a feeling it's deeper than just Atom, but something within Electron.

**Comment by rainstormza** [comment]
`brew cask install atom`

**Comment by Superpencil** [comment]
This is an Electron issue I think

**Comment by Bl4Cc4t** [comment]
@rainstormza
Worked like a charm.
Thanks a bunch, didn’t know atom is in the cask!

**Comment by MarqueIV** [comment]
I think I found the issue.  It has to do with apps downloaded from the internet.  In the case of normal installers, it prompts you to verify it at the point of install.  But VS Code is an application you manually extract from a zip file, so there is no installer.  That prompt comes the first time you try and run it.

The problem is most people just drag the app to Applications, then to the dock, *then* run it.  But you've dragged an app to the dock before it's been verified which is the real cause of the problem.  The solution is to verify it *before* you add it to the dock.

To be sure this was the issue, I deleted it from my dock, then from Applications, then using the exact same ZIP file I already had downloaded, I again extracted the app, moved it to Applications, then added it to the dock, *then* launched it, and sure enough, it was back to the double-icon/icon disappearing behavior.

I again deleted everything, but this time after I unzipped it into Downloads, it verified and opened fine.  I then copied it to Applications, then the dock and it worked as expected.

As a third test, I repeated test 2, but this time copied it to Applications first, before running it, then ran it from there, again before adding it to the dock. Same thing. It worked.

TLDR version, it's not a VS Code issue, it's not an Electron issue.  It's not an Archive Utility vs Dr. Archiver or anything else issue.  It's a 'You added it to the dock before you verified it was safe!' issue.  Before you add it to the dock, open and run it (from anywhere. Doesn't matter.) then you can safely add it to the dock.  This seems it would be the case for *any* zip-distributed app.

Hope this helps!

**Comment by stevenreddie** [comment]
For anyone having this issue, see my root cause and workaround that I just posted at https://github.com/electron/electron/issues/9293#issuecomment-331422359

**Comment by juanencalada** [comment]
@MarqueIV 

Thank you. This worked for me.

**Comment by Halordain** [comment]
This shouldn't require a workaround; other apps in Sierra don't suffer from this problem. It's annoying to users to have to read a GitHub post and type commands into the terminal, and for many people, they expect it to "just work" like other apps and text editors.  It can be a dealbreaker (i.e. unusable) when they can't reliably keep Atom in the dock.

**Comment by MarqueIV** [comment]
Please read my post.  This isn't a problem with VS Code or Electron or anything else. It's a case of a not-yet-verified app being added to the dock.  Again, TLDR version of my original comment above, before you add it to the dock, open it and answer the 'app downloaded from the internet' prompt.  Once you've affirmed that you do want to open it, you can safely add it to the dock.  This is the case with *any* app that would put up that 'downloaded from the internet' prompt.  Any.

**Comment by MarqueIV** [comment]
So just to confirm, you are saying that you downloaded and unzipped the app, launched it where it was, answered the “downloaded from the Internet“ prompt in the affirmative, closed the app, then dragged it to the dock, and it still is exhibiting that behavior?
